salt/proxy/junos.py

```diff
@@ -118,10 +120,18 @@ def conn():
 
 def alive(opts):
     '''
-    Return the connection status with the remote device.
+    Validate and return the connection status with the remote device.
 
     .. versionadded:: Oxygen
     '''
+    try:
+        thisproxy['conn'].rpc('<file-list><path>/dev/null</path></file-list>')
+
+    except RpcTimeoutError:
+        thisproxy['conn'].connected = False
+    except ConnectClosedError:
+        thisproxy['conn'].connected = False
```

Comment:
@vnitinv  is correct that this is unnecessary. `connected` is already set to `False` for you in this case.
https://github.com/Juniper/py-junos-eznc/blob/master/lib/jnpr/junos/exception.py#L300